core/src/main/java/org/elasticsearch/search/aggregations/bucket/range/geodistance/GeoDistanceParser.java

```diff
@@ -164,6 +162,11 @@ public AggregatorFactory parse(String aggregationName, XContentParser parser, Se
         return new GeoDistanceFactory(aggregationName, vsParser.config(), InternalGeoDistance.FACTORY, origin, unit, distanceType, ranges, keyed);
     }
 
+    private static String getSearchParseExceptionMessage(String currentFieldName, String aggregationName) {
+        return "Failed when parsing field: '" +  currentFieldName +
```

Comment:
Those token names are important, as they tell you what type of JSON token was expected.  They may not make sense  to you yet, but they are invaluable for figuring out what to look for, eg I was expecting `VALUE_STRING` but got `START_OBJECT`  (ie i was expecting `"something"` and got `{`)